packages/core/src/dispatcher.ts

```diff
@@ -58,8 +62,70 @@ export default class Dispatcher extends EventEmitter {
    */
   async init(): Promise<void> {
     this._peerId = this._peerId || (await this._ipfs.id()).id
-    await this._ipfs.pubsub.subscribe(this.topic, this.handleMessage.bind(this))
-    this._log({ peer: this._peerId, event: 'subscribed', topic: this.topic })
+    await this._subscribe()
+    !TESTING && this._resubscribeOnDisconnect()
+  }
+
+  /**
+   * Subscribes IPFS pubsub to `this.topic` and logs an `subscribe` event.
+   */
+  async _subscribe(): Promise<void> {
+    if (this._isSubscribed) {
+      console.log(`Pubsub subscribe skipped. Already subscribed to topic ${this.topic}`)
+    } else {
+      try {
+        await this._ipfs.pubsub.subscribe(
+          this.topic,
+          this.handleMessage.bind(this),
+          {timeout: !TESTING && IPFS_GET_TIMEOUT}
+        )
+
+        const { isSubscribed, error } = await this._confirmIsSubscribed()
+        if (isSubscribed) {
+          this._log({peer: this._peerId, event: 'subscribed', topic: this.topic })
+        } else if (error) {
+          throw error
+        } else {
+          throw new Error(`Pubsub subscribe failed for topic ${this.topic}`)
+        }
+      } catch (error) {
+        console.error(error)
```

Comment:
Can you add a comment `// TODO use logger`?  I've already left a few TODOs throughout the codebase with places to log once we have the logging framework in place, so that'll make sure I remember to include this in my logging pass

Reply:
Sure